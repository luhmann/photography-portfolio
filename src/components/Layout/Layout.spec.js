import React from 'react';
import { render } from 'react-testing-library';
import generateProps from 'react-generate-props';

import { LayoutComponent } from './';

const createTestProps = (custom = {}) => ({
  ...generateProps(LayoutComponent, { optional: true }),
  ...custom,
});

test('should render as expected - when on home-page', () => {
  const props = createTestProps({
    location: {
      pathname: '/',
    },
  });
  const { container } = render(<LayoutComponent {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});

test('should render as expected - when not home-page', () => {
  const props = createTestProps();
  const { container } = render(<LayoutComponent {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
