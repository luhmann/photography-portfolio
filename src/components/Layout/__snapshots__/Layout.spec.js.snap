// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected - when not home-page 1`] = `
<main
  class="sc-bdVaJa cpfohf"
>
  <a
    class="sc-bZQynM iMHTUI"
    color="black"
    href="/"
  >
    <h1
      class="sc-htpNat iiKRXJ"
    >
      JF Dietrich Photography
    </h1>
  </a>
  <a
    class="sc-bxivhb bugtHR"
  >
    menu
  </a>
  <nav
    class="sc-ifAKCX dQsBdt"
    data-testid="menu-slideout"
  >
    <div
      class="sc-EHOje kQpzht"
      data-testid="menu-album"
    >
      albumTitle
    </div>
    <a
      class="sc-bZQynM ADRzZ"
      color="black"
      data-testid="menu-gallery"
      display="block"
      font-family="body"
      font-size="m,l"
      font-weight="bold"
      href="path"
      mb="4"
    >
      title
    </a>
  </nav>
  children
</main>
`;

exports[`should render as expected - when on home-page 1`] = `
<main
  class="sc-bdVaJa cpfohf"
>
  children
</main>
`;
